addons/sale/models/sale_order_line.py

```diff
@@ -703,6 +703,9 @@ def _compute_qty_delivered(self):
         mapping = lines_by_analytic._get_delivered_quantity_by_analytic([('amount', '<=', 0.0)])
         for so_line in lines_by_analytic:
             so_line.qty_delivered = mapping.get(so_line.id or so_line._origin.id, 0.0)
+        # add a default value of 0.0 to other records if necessary
+        for so_line in self - lines_by_analytic:
+            so_line.qty_delivered = so_line.qty_delivered or 0.0
```

Comment:
what about `default=0` in the field definition ?

Reply:
That was also my initial call but I was told that the default value of a computed field should rather been set by its compute method. Unfortunately I can not find this info in the documentation but if I remember correctly @nle-odoo knew at least one reason why.

Reply:
![](https://c.tenor.com/OgJLaotHGYEAAAAd/tenor.gif)

Reply:
![oss117-faisonscommeca](https://github.com/odoo/odoo/assets/153811290/9e58d80f-e8bc-4bf9-b989-fb38430af089)

Reply:
Hello @Whenrow this has been changed accordingly !

Reply:
And I have added a test 😊